meilisearch/src/search.rs

```diff
@@ -281,12 +292,17 @@ pub fn add_search_rules(query: &mut SearchQuery, rules: IndexSearchRules) {
 pub fn perform_search(
     index: &Index,
     query: SearchQuery,
+    features: RoFeatures,
 ) -> Result<SearchResult, MeilisearchHttpError> {
     let before_search = Instant::now();
     let rtxn = index.read_txn()?;
 
     let mut search = index.search(&rtxn);
 
+    if let Some(ref vector) = query.vector {
+        search.vector(vector.clone());
+    }
+
     if let Some(ref query) = query.q {
         search.query(query);
```

Comment:
shouldn't adding both `"q"` and `"vector"` to a query be an error?

My understanding is that the current behavior is silently ignoring the `"q"` part, but isn't there a risk of a user thinking they are using hybrid search, which is not currently implemented?

not necessarily to be fixed in this PR though.

Reply:
I just printed a warning and edited the feature discussion https://github.com/meilisearch/product/discussions/677